zpeclient.go

```diff
@@ -300,3 +335,45 @@ func (c *Cache) authorize(principal string, check AthenzAccessCheck) (bool, erro
 	}
 	return false, nil
 }
+
+// parseUpdateTime - read the last update time as string from configmap, store in the cache
+func (c *Cache) parseUpdateTime(configmap interface{}) error {
+	var lastUpdateTime string
+
+	obj, ok := configmap.(*corev1.ConfigMap)
+	if !ok {
+		return fmt.Errorf("Unable to cast interface to config map object")
+	}
+
+	// check if lastUpdateKeyField exists in the configmap
+	if _, ok = obj.Data[lastUpdateKeyField]; ok {
+		lastUpdateTime = strings.ReplaceAll(obj.Data[lastUpdateKeyField], `"`, "")
+	} else {
+		return fmt.Errorf("configmap format is wrong, it doesn't have a field called: %v", lastUpdateKeyField)
+	}
+	// update cache status boolean in the cache object
+	return c.updateCacheStatus(lastUpdateTime)
+}
+
+// updateCacheStatus - read lastUpdate variable, compare with current time and update on cache status
+func (c *Cache) updateCacheStatus(timestamp string) error {
+	c.cmLock.Lock()
+	defer c.cmLock.Unlock()
+	var err error
+	if timestamp != "" {
+		c.lastUpdate, err = time.Parse(time.RFC3339Nano, timestamp)
```

Comment:
if there was an error with time.Parse() would the `c.lastUpdate` get updated? Subsequently can lead to incorrect processing here:```diff := t.Sub(c.lastUpdate)```

Reply:
if timestamp format is wrong, time.Parse() returns default value `0001-01-01 00:00:00 +0000 UTC`. It will result in incorrect processing, let me update the logic.